common-custom-user-data-gradle-plugin/src/main/java/com/gradle/Utils.java

```diff
@@ -116,7 +116,7 @@ static Properties readPropertiesFile(String name, ProviderFactory providers, Gra
     }
 
     static InputStream readFile(String name, ProviderFactory providers, Gradle gradle) throws FileNotFoundException {
-        if (isGradle65OrNewer()) {
+        if (isGradle65OrNewer() && providers != null && gradle != null) {
```

Comment:
I believe these null checks can be removed, given your latest changes.

Reply:
Good catch! I had removed the other ones, but I missed these.